rasa/shared/core/events.py

```diff
@@ -1488,6 +1489,7 @@ def __init__(
         metadata: Optional[Dict] = None,
         action_text: Optional[Text] = None,
         hide_rule_turn: bool = False,
+        predicted_action_unlikely_intent: bool = False,
```

Comment:
We need engineering advice here. My take is that we should try to avoid leaking the logic of what happened in the conversation in the past into `ActionExecuted`.
@alwx's [take](https://rasa-hq.slack.com/archives/C8LEHMTAM/p1625828747322600?thread_ts=1625644380.222100&cid=C8LEHMTAM) is that it's quite complex / hacky to implement the functionality we need without doing this.

Reply:
@alwx You mentioned that this is done to handle the case when:
1. `action_unlikely_intent` was predicted.
2. It wasn't expected to be predicted.
3. Ensemble still predicted the correct action just after `action_unlikely_intent`.

So, to display cases like this:

```
- action: any_correct_action # predicted: action_unlikely_intent
```

Can't we create a new event `WarningPredicted` which also inherits from `ActionExecuted` and use it to handle this case?

Reply:
I'll be in meetings for the next couple of hours. Can you also please take a look at why `tests/shared/core/test_trackers.py::test_tracker_write_to_story` is failing? I think it's because of the new attribute added and other parts of the codebase not expecting it. (My original worry 😅 )

Reply:
Done!